src/layouts/primary-layout.js

```diff
@@ -20,76 +20,52 @@ import ProfilePage from '../pages/profile-page'
 import SelectionProcessPage from '../pages/selection-process-page'
 import TracksGuidePage from '../pages/tracks-guide-page'
 import PresentationLayout from './presentation-layout'
-import { getAllFromSummit } from '../actions/base-actions';
+import {getSelectionPlan} from "../actions/base-actions";
 
-class PrimaryLayout extends React.Component {
+const PrimaryLayout = ({location, summit, speaker, member, match, selectionPlan, getSelectionPlan}) => {
+    const loggedUser = (speaker && speaker.id) ? speaker : member;
+    const selectionPlanIdParam = parseInt(match.params.selection_plan_id);
 
-    componentDidMount() {
-        let summitSlug = this.props.match.params.summit_slug;
-        this.props.getAllFromSummit(summitSlug);
-    }
-
-    componentWillReceiveProps(newProps) {
-        let oldSummitSlug = this.props.match.params.summit_slug;
-        let newSummitSlug = newProps.match.params.summit_slug;
-
-        if (newSummitSlug && newSummitSlug !== oldSummitSlug) {
-            this.props.getAllFromSummit(newSummitSlug);
+    const getActiveMenu = () => {
+        if (location.pathname.includes('presentations')) {
+            return 'presentations';
         }
-    }
 
-    getActiveMenu() {
-        let {location, summit} = this.props;
-        switch(location.pathname) {
-            case `/app/${summit.slug}/presentations`:
-                return 'presentations';
-                break;
-            case `/app/${summit.slug}/profile`:
-                return 'profile';
-                break;
+        if (location.pathname.includes('profile')) {
+            return 'profile';
         }
-    }
-
-    render(){
-        let { location, speaker, member, summit, loading, match, baseLoaded } = this.props;
-
-        if (!summit || !baseLoaded) return null;
+    };
 
-        if(summit.slug !== match.params.summit_slug) return null;
-
-        if((!speaker || !speaker.id) && location.pathname !== `/app/${summit.slug}/profile` && !loading) {
-            return (
-                <Redirect exact to={{ pathname: `/app/${summit.slug}/profile` }}  />
-            );
+    useEffect(() => {
+        if (selectionPlan?.id !== selectionPlanIdParam) {
+            getSelectionPlan(summit.id, selectionPlanIdParam);
         }
+    }, [selectionPlan, match]);
```

Comment:
could we use more granular deps ?
like 
[selectionPlan?.Id, summit?.id, selectionPlanIdParam]

Reply:
we don't need summit.id here , I'll add the rest